src/main/java/edit_user_public_profile_usecase/EditUserPublicProfileInteractor.java

```diff
@@ -0,0 +1,80 @@
+package edit_user_public_profile_usecase;
+
+import Entities.User;
+
+/**
+ * The edit user's public profile use case.
+ */
+public class EditUserPublicProfileInteractor implements EditUserPublicProfileInputBoundary {
+    final private EditUserPublicProfileDSGateway userDSGateway;
+    final private EditUserPublicProfileOutputBoundary profileOutputBoundary;
+
+    public EditUserPublicProfileInteractor(EditUserPublicProfileDSGateway userDSGateway,
+                                           EditUserPublicProfileOutputBoundary profileOutputBoundary) {
+        this.userDSGateway = userDSGateway;
+        this.profileOutputBoundary = profileOutputBoundary;
+    }
+
+    /**
+     *
+     * @param requestModel the request model for the edit user's public profile use case.
+     * @return the new edited changes to the public profile.
+     */
+    @Override
+    public EditUserPublicProfileResponseModel saveEdits(EditUserPublicProfileRequestModel requestModel) {
+        /*Creating a failed response model*/
+        EditUserPublicProfileResponseModel profileFailedResponseModel = new EditUserPublicProfileResponseModel(
+                requestModel.getPreferences(),
+                requestModel.getCoursePreferences(),
+                requestModel.getBio(),
+                "");
+
+        /*Returning a failed view when preferences are left blank*/
+        for (String p: requestModel.getPreferences().values()) {
+            if (p.equals("")) {
+                profileFailedResponseModel.setMessage("Preferences was left blank.");
+                return profileOutputBoundary.prepareFailView(profileFailedResponseModel);
+            }
+        }
+
+        /*Returning a failed view when preferences are left blank*/
+        if (requestModel.getCoursePreferences().equals("")) {
+            profileFailedResponseModel.setMessage("Course preferences was left blank.");
+            return profileOutputBoundary.prepareFailView(profileFailedResponseModel);
+        }
+
+        /*Returning a failed view when profile bio is left blank*/
+        if (requestModel.getBio().equals("")) {
+            profileFailedResponseModel.setMessage("Biography was left blank.");
+            return profileOutputBoundary.prepareFailView(profileFailedResponseModel);
+        }
+
```

Comment:
It might be easier if you had your prepareFailView have only an input of a string. I mentioned this in another comment, but a Response model is not necessary for input as we will be throwing an error with the specified string. At least from my understanding, you do not need all the User's profile information for a fail view, just the message. It will also make your code less redundant.
Maybe I am misunderstanding something.